app/forms/CommodityCodeDigitsFormProvider.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+class CommodityCodeDigitsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "value" -> text("commodityCodeDigits.error.required")
+        .verifying(maxLength(100, "commodityCodeDigits.error.length"))
```

Comment:
Worth checking with Matt or Joe but the max length of a commodity code is more like 22 digits. Should we verify . that it's numeric also?

Reply:
Have loosely discussed input limitations with Aimee. I know the address inputs have limitations due to EU law and address system compliance i.e. 4 line address and so on.

Reply:
I'd argue that a story isn't complete until we at least have an initial idea of what these limitations should be. In the operational ui we have gone with 22 Digits as the restriction for Commodity Codes as decided by Joe M.